### DISCUSSION TITLE
20220802

### DISCUSSION THREAD

**Discussion by tmiw** [ARSFI/paclink-open-source#32] [General]
This is a pre-production build intended to gauge current status of the project and determine what needs to be fixed. Brief testing indicates that packet (using Kenwood TM-D700A), AGW (using AGWPE and UZ7HO soundmodem), Telnet and Pactor channels work.

Contains (vs. [last pre-release version](https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220719)):

* Remove forced capitalization of the password for POP3 and SMTP logins.

Instructions:

1. Unzip paclink-open-source.zip somewhere on your hard drive.
2. Run setup.exe to install on your system. Paclink will be installed in the Start menu.
3. Open Paclink from the Start menu and reconfigure if required. (There is no import functionality from the existing Paclink at this time but if you used a pre-release version, configuration should be saved.)

Please leave comments in the release's discussion thread or email K6AQ@winlink.org with feedback. 👍
 

<hr /><em>This discussion was created from the release <a href='https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220802'>20220802</a>.</em>

**Comment by kd4vrf** [comment]
I've still noticed some strange behavior on the passwords so I dug into it over the last week or so and here is what I found:

When you change the POP3 password it doesn't take affect immediately which lead to some confusion on my part. When changing the password in the "Manage Callsign Accounts" the change is only applied after you reboot Paclink. If this is the intended behavior I think it would be helpful put a note somewhere in the dialog box to call that out.

The POP3 password is also available to edit in the "Packlink Site Properties" dialog box. However, when looking at the "Packlink Site Properties" dialog box:

- any changes made in the "Manage Callsign Accounts" dialog box aren't reflected in the "Password (for POP3/SMTP):" field on the "Packlink Site Properties" dialog box (e.g. when you change the password in "Manage Callsign Accounts" and reboot, when you open the "Packlink Site Properties" dialog box the "Password (for POP3/SMTP):" field is unchanged even though you can use the new password to login to the POP3 server in Paclink).
- any changes made in the "Packlink Site Properties" don't seem to have any affect on the password, even after a reboot
- the "Password (for POP3/SMTP):" field is still case sensitive and forced to upper case

Screenshot of the "Packlink Site Properties" dialog box showing upper case even though the password is in lower case:
![Paclink Site Properties - Screen Shot 2022-09-02 at 10 30 32 AM](https://user-images.githubusercontent.com/45251519/188177645-f36c2f0f-26a4-4f1c-9fb0-59cb8293b347.png)

Screenshot of the "Manage Callsign Accounts" dialog box showing the proper password and case:
![Manage Callsign Accounts - Screen Shot 2022-09-02 at 10 34 15 AM](https://user-images.githubusercontent.com/45251519/188177772-03cc1a0c-efe2-48cb-bccc-6425f1dd731b.png)

Screenshot of changing the POP3 password in the "Manage Callsign Accounts" dialog box and showing the "Packlink Site Properties" dialog box not reflecting the changes after a reboot (Changed the password to test12345 and the "Packlink Site Properties" still shows test1234):
![Paclink Site Properties after password change - Screen Shot 2022-09-02 at 11 03 09 AM](https://user-images.githubusercontent.com/45251519/188178648-88bbb2eb-5f41-4ece-8650-3d37342ea633.png)

**Comment by k0qed** [comment]
If you want to assign this to me I'll dig into it when I get back from my
camping trip in a little over a week from now.

Lee, K0QED

**Reply by tmiw** [reply]
Go for it. 👍

**Comment by k0qed** [comment]
I'm committing some changes to address issues with passwords. The Manage
Callsign Accounts dialog was confusing since it referenced the POP3/SMTP
password. It did nothing with that password - it only managed the callsign
account's password. I cleaned that up and fixed an issue where the
callsign account password was being forced to uppercase.

FYI: In our Winlink meeting Wednesday we talked about this project and how
we might elicit additional contributors. We'll be trying to get an
announcement in the ARRL newsletter and possibly QST. It will
briefly describe the project and point them to the repository.

-Lee, K0QED

**Reply by tmiw** [reply]
Thanks! [I created a new release](https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20221011) with those changes. @kd4vrf, can you give it a shot?

As for finding additional people, that would be nice. I do know there's been some interest but there were challenges with environment setup. We may need to get it to the point where it can minimally run on e.g. Linux (say, by bringing up the minimum required for packet to work).
